pkg/sql/sem/tree/type_check.go

```diff
@@ -520,7 +520,9 @@ func onCastTypeCheckHook(from, to types.Family) {
 		return
 	}
 	panic(errors.AssertionFailedf(
-		"no cast counter found for cast from %s to %s", from.Name(), to.Name(),
+		"no cast counter found for cast from %s to %s",
+		// Family names are always safe for redacting.
```

Comment:
given this comment, it would have been a bit nicer to update `func (f Family) Name()` so it returns a `redact.SafeString`, so the same change doesn't need to be made at all call sites